addons/calendar/models/calendar_event.py

```diff
@@ -286,16 +286,11 @@ def _compute_attendees_count(self):
     @api.depends('partner_ids')
     @api.depends_context('uid')
     def _compute_user_can_edit(self):
-        for event in self:
-            # By default, only current attendees and the organizer can edit the event.
-            editor_candidates = set(event.partner_ids.user_ids + event.user_id)
-            # Right before saving the event, old partners must be able to save changes.
-            if event._origin:
-                editor_candidates |= set(event._origin.partner_ids.user_ids)
-            # Non-private events must be editable by uninvited administrators.
-            if self.env.user.has_group('base.group_system') and event.privacy != 'private':
-                editor_candidates.add(self.env.user)
-            event.user_can_edit = self.env.user in editor_candidates
+        self.user_can_edit = False
+        new_events = self.filtered(lambda ev: isinstance(ev.id, models.NewId))
+        new_events.user_can_edit = True
+        edit_events = (self - new_events)._filtered_access('write')
+        edit_events.user_can_edit = True
```

Comment:
Suggestion to avoiding writing twice in events from (self ∩ new_events) and (self ∩ edit_events):

```suggestion
        new_events = self.filtered(lambda ev: isinstance(ev.id, models.NewId))
        edit_events = (self - new_events)._filtered_access('write')
        editable_events = new_events + edit_events
        editable_events.user_can_edit = True
        (self - editable_events).user_can_edit = False
```